packages/multicast_dns/lib/src/resource_record.dart

```diff
@@ -338,11 +338,15 @@ class TxtResourceRecord extends ResourceRecord {
   const TxtResourceRecord(
     String name,
     int validUntil, {
-    required this.text,
+    required this.attrs,
```

Comment:
Apologies for the delay in this review.

I believe this would be a breaking change to [TxtResourceRecord](https://pub.dev/documentation/multicast_dns/latest/multicast_dns/TxtResourceRecord/TxtResourceRecord.html)?

How about a new named constructor:

```dart
  TxtResourceRecord.attributes(
    String name,
    int validUntil, {
    required List<String> attributes,
  }) : this(name, validUntil, text: attributes.join('\n'));
```
Also total nit but `attributes` instead of `attrs`?

Reply:
A way to do this without it being a breaking change would be to:
1. Make `text` nullable and mark it deprecated.
2. Add `attrs` as nullable.
3. Throw if (a) both or (b) neither are null.

After the deprecation period, mark `attrs` as non-nullable and remove `text`.

If very few packages are relying on this, then it might not be worth the effort.